### PR TITLE
Clarify the description of SUBSTRING and its legacy mode setting.

### PR DESCRIPTION
**Pull request by miguno**

### CODE REVIEW

**Comment by miguno** [comment]
As discussed offline with @rodesai.

Also: Is there any function other than `SUBSTRING` -- like our functions to work with arrays -- where we changed from indexing with start position 0 to indexing with start position 1?  If so, then we need another update PR for the docs.

File: docs/developer-guide/syntax-reference.rst [hunk 1]
```diff
@@ -1190,20 +1190,27 @@ Scalar functions
...
+|                        |                                                                           | For example, ``SUBSTRING("stream", 1, 4)`` would  |
+|                        |                                                                           | return "tre".                                     |
 |                        |                                                                           | It is possible to switch back to this legacy mode |
 |                        |                                                                           | by setting                                        |
 |                        |                                                                           | ``ksql.functions.substring.legacy.args`` to       |
-|                        |                                                                           | ``true``. Though this is not recommended.         |
+|                        |                                                                           | ``true``. However, we recommend to not enable     |
```
Comment (JimGalasyn) [inline docs/developer-guide/syntax-reference.rst]: "We recommend against enabling..."

File: docs/developer-guide/syntax-reference.rst [hunk 2]
```diff
@@ -1190,20 +1190,27 @@ Scalar functions
...
+|                        |                                                                           | this setting, if possible, and instead update     |
```
Comment (JimGalasyn) [inline docs/developer-guide/syntax-reference.rst]: "this setting. Instead, update..."

**Comment by miguno** [comment]
Updated, @JimGalasyn.  Will merge.

**Pull request merged**